queue/driver.go

```diff
@@ -15,8 +15,13 @@ type remoteQueueDriver interface {
 	Open(context.Context) error
 	Close()
 
-	// Get finds a job by job ID.
+	// Get finds a job by job ID. For retryable jobs, this returns the latest
+	// job attempt.
```

Comment:
Would it make sense to have another `Get` that gets the latest job attempt and returns an `amboy.RetryableJob`. More of a convenience function than anything, but would reduce all the type checking.

Reply:
I feel like that would be annoying because you can't overload method names unless I made a `GetLatest` on the `RetryableQueue` which behaves identically to `Get()`. There's only one place that depends on `Get()` returning a retryable job.

Reply:
Sounds good.